Gems/AtomLyIntegration/CommonFeatures/Code/Source/ImageBasedLights/EditorImageBasedLightComponent.cpp

```diff
@@ -75,5 +83,74 @@ namespace AZ
         {
         }
 
+
+        AZ::u32 EditorImageBasedLightComponent::OnDiffuseImageAssetChanged()
+        {
+            ImageBasedLightComponentConfig& configuration = m_controller.m_configuration;
+
+            if (UpdateImageAsset(configuration.m_diffuseImageAsset, "_ibldiffuse", "Diffuse",
+                                 configuration.m_specularImageAsset, "_iblspecular", "Specular"))
+            {
+                m_controller.SetSpecularImageAsset(configuration.m_specularImageAsset);
+            }
+
+            return AZ::Edit::PropertyRefreshLevels::ValuesOnly;
+        }
+
+        AZ::u32 EditorImageBasedLightComponent::OnSpecularImageAssetChanged()
+        {
+            ImageBasedLightComponentConfig& configuration = m_controller.m_configuration;
+
+            if (UpdateImageAsset(configuration.m_specularImageAsset, "_iblspecular", "Specular",
+                                 configuration.m_diffuseImageAsset, "_ibldiffuse", "Diffuse"))
+            {
+                m_controller.SetDiffuseImageAsset(configuration.m_diffuseImageAsset);
+            }
+
+            return AZ::Edit::PropertyRefreshLevels::ValuesOnly;
+        }
+
+        bool EditorImageBasedLightComponent::UpdateImageAsset(
+            Data::Asset<RPI::StreamingImageAsset>& asset1,
+            const char* asset1Suffix,
+            const char* asset1Name,
+            Data::Asset<RPI::StreamingImageAsset>& asset2,
+            const char* asset2Suffix,
+            const char* asset2Name)
+        {
+            AZStd::string assetPath = asset1.GetHint();
+            AZ::Data::AssetId assetId;
+
+            if (!assetPath.empty())
+            {
+                // try to load the matching image asset by replacing the asset1 suffix with the asset2 suffix
+                AZ::StringFunc::Replace(assetPath, asset1Suffix, asset2Suffix);
+
+                assetId = AZ::RPI::AssetUtils::GetAssetIdForProductPath(assetPath.c_str(), AZ::RPI::AssetUtils::TraceLevel::None);
+                if (!assetId.IsValid())
+                {
+                    // unable to locate the matching image asset
+                    return false;
+                }
+            }
+
+            if (asset2.Get())
```

Comment:
Check against the asset ID instead of the asset pointer in case the asset was never queued or loaded.

```suggestion
            if (asset2.GetId().IsValid())
```

Reply:
Done.